src/Umbraco.Web/Editors/ParameterSwapControllerActionSelector.cs

```diff
@@ -25,33 +35,92 @@ public ParameterSwapControllerActionSelector(params ParameterSwapInfo[] actions)
         {
             _actions = actions;
         }
+
         public override HttpActionDescriptor SelectAction(HttpControllerContext controllerContext)
         {
             var found = _actions.FirstOrDefault(x => controllerContext.Request.RequestUri.GetLeftPart(UriPartial.Path).InvariantEndsWith(x.ActionName));
 
             if (found != null)
             {
-                var id = HttpUtility.ParseQueryString(controllerContext.Request.RequestUri.Query).Get(found.ParamName);
+                HttpActionDescriptor method;
+                if (TryBindFromUri(controllerContext, found, out method))
+                {
+                    return method;
+                }
 
-                if (id != null)
+                //if it's a post we can try to read from the body and bind from th
+                if (controllerContext.Request.Method == HttpMethod.Post)
                 {
-                    var idTypes = found.SupportedTypes;
+                    var requestContent = new HttpMessageContent(controllerContext.Request);
+                    var strJson = requestContent.HttpRequestMessage.Content.ReadAsStringAsync().Result;
+                    var json = JsonConvert.DeserializeObject<JObject>(strJson);
 
-                    foreach (var idType in idTypes)
+                    if (json == null)
                     {
-                        var converted = id.TryConvertTo(idType);
-                        if (converted)
+                        return base.SelectAction(controllerContext);
+                    }
+
+                    var requestParam = json[found.ParamName];
+
+                    if (requestParam != null)
+                    {
+                        var paramTypes = found.SupportedTypes;
+
+                        foreach (var paramType in paramTypes)
                         {
-                            var method = MatchByType(idType, controllerContext, found);
-                            if (method != null)
-                                return method;
+                            try
+                            {
+                                var converted = requestParam.ToObject(paramType);
+                                if (converted != null)
+                                {
+                                    method = MatchByType(paramType, controllerContext, found);
+                                    if (method != null)
+                                        return method;
+                                }
+                            }
+                            catch (JsonReaderException)
+                            {
+                                //can't convert
```

Comment:
Should we log this out to the logfile, rather than swallow the exception silently that this method was trying to be bound to the correct API method in the controller but failed.